File: backend/src/main/java/com/microfinance/code/dto/LoanDashboardDTO.java
package com.microfinance.code.dto;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
public class LoanDashboardDTO {
    private BigDecimal netCashFlow;
    private BigDecimal totalDisbursements;
    private BigDecimal totalRepayments;
    private BigDecimal outstandingPortfolio;
    private BigDecimal delinquencyRate;

    // Breakdowns
    private BigDecimal smeDisbursements;
    private BigDecimal hpDisbursements;
    private BigDecimal smeRepayments;
    private BigDecimal hpRepayments;
    private LocalDate startDate;
    private LocalDate endDate;
}

File: frontend/MicroFinance_APP/src/app/model/LoanDashboardDTO.ts
import { Observable } from "rxjs";

export interface LoanDashboardDTO {
    netCashFlow: number;
    totalDisbursements: number;
    totalRepayments: number;
    outstandingPortfolio: number;
    delinquencyRate: number;
    smeDisbursements: number;
    hpDisbursements: number;
    smeRepayments: number;
    hpRepayments: number;
    startDate: string;
    endDate: string;
  }

  export interface LoanDashboardService {
    getLoanMetrics(startDate: string, endDate: string): Observable<LoanDashboardDTO>;
  }